docs/book/intro.md

```diff
@@ -1,3 +1,132 @@
 # zend-mvc-middleware
 
-This component provides ...
+This library provides the ability to dispatch middleware pipelines in place of
+controllers within zend-mvc.
+
+## Dispatching PSR-7 Middleware
+
+[PSR-7](http://www.php-fig.org/psr/psr-7/) defines interfaces for HTTP messages,
+and is now being adopted by many frameworks; Zend Framework itself offers a
+parallel microframework targeting PSR-7 with [Expressive](https://docs.zendframework.com/zend-expressive).
+What if you want to dispatch PSR-7 middleware from zend-mvc?
+
+zend-mvc currently uses [zend-http](https://github.com/zendframework/zend-http)
+for its HTTP transport layer, and the objects it defines are not compatible with
+PSR-7, meaning the basic MVC layer does not and cannot make use of PSR-7
+currently.
+
+However, starting with version 2.7.0, zend-mvc offers
+`Zend\Mvc\MiddlewareListener`. This `Zend\Mvc\MvcEvent::EVENT_DISPATCH`
+listener listens prior to the default `DispatchListener`, and executes if the
+route matches contain a "middleware" parameter, and the service that resolves to
+is callable. When those conditions are met, it uses the [PSR-7 bridge](https://github.com/zendframework/zend-psr7bridge)
+to convert the zend-http request and response objects into PSR-7 instances, and
+then invokes the middleware.
+
+Starting with zend-mvc version 3.2.0, `Zend\Mvc\MiddlewareListener` is deprecated and replaced
+by `Zend\Mvc\Middleware\MiddlewareListener` provided by this package.  
+After package installation, `Zend\Mvc\Middleware` module must be registered in your
+zend-mvc based application.
+
+## Mapping routes to middleware
```

Comment:
For all headlines:

* a noun must be start with a capital letter
* all other words are written in small letters